pyanalyze/implementation.py

```diff
@@ -352,20 +352,20 @@ def inner(iterable: Value) -> Value:
 def _list_append_impl(ctx: CallContext) -> ImplReturn:
     lst = replace_known_sequence_value(ctx.vars["self"])
     element = ctx.vars["object"]
-    varname = ctx.visitor.varname_for_self_constraint(ctx.node)
-    if varname is not None:
-        if isinstance(lst, SequenceValue):
+    if isinstance(lst, SequenceValue):
+        varname = ctx.visitor.varname_for_self_constraint(ctx.node)
+        if varname is not None:
             no_return_unless = Constraint(
                 varname,
                 ConstraintType.is_value_object,
                 True,
                 SequenceValue.make_or_known(list, (*lst.members, (False, element))),
             )
             return ImplReturn(KnownValue(None), no_return_unless=no_return_unless)
-        elif isinstance(lst, GenericValue):
-            return _check_generic_container(
-                "list.append", "object", ctx.vars["self"], lst, element, ctx, list
-            )
+    if isinstance(lst, GenericValue):
```

Comment:
This could alternatively be elif, which would allow us to append anything to a SequenceValue (i.e., unannotated) list. But I just ran this branch internally and only got three errors, all of which were fixable by adding explicit annotations. I think such annotations are helpful, so I'll stick with this logic.